Controller/CRUDController.php

```diff
@@ -1080,7 +1080,9 @@ protected function redirectTo($object)
 
         if (!$url) {
             foreach (array('edit', 'show') as $route) {
-                if ($this->admin->hasRoute($route) && $this->admin->isGranted(strtoupper($route), $object)) {
+                if ($this->admin->hasRoute($route) 
+                    && $this->admin->isGranted(strtoupper(str_replace('show', 'view', $route)), $object)
```

Comment:
In fact, you might want to do `hasAccess`:

https://github.com/sonata-project/SonataAdminBundle/blob/3.x/Admin/AbstractAdmin.php#L2833

`checkAccess` will throw an exception, and that's not what we want here